cpp/open3d/t/geometry/TriangleMesh.cpp

```diff
@@ -1215,6 +1215,139 @@ TriangleMesh TriangleMesh::SelectByIndex(const core::Tensor &indices) const {
+            point_to_old_index[coord] = i;
+            index_old_to_new[i] = k;
+            ++k;
+            vertex_mask_ptr[i] = 1;
+        } else {
+            index_old_to_new[i] = index_old_to_new[point_to_old_index[coord]];
+        }
+    }
+    vertex_mask = vertex_mask.To(mesh.GetDevice(), core::Bool);
+
+    //REM_DUP_VERT_STEP 2:
+    // Update vertex attributes, by shrinking them appropriately.
+    for (auto item : mesh.GetVertexAttr()) {
+        mesh.SetVertexAttr(item.first, item.second.IndexGet({vertex_mask}));
+    }
+
+    //REM_DUP_VERT_STEP 3:
+    // Remap triangle indices to new unique vertex indices, if required.
+    if (k < orig_num_vertices) {
+        core::Tensor tris = mesh.GetTriangleIndices();
+        core::Tensor tris_cpu = tris.To(core::Device()).Contiguous();
+        const int64_t num_tris = tris_cpu.GetLength();
+        //Update triangle indices.
+        Tindex_t *indices_ptr = tris_cpu.GetDataPtr<Tindex_t>();
+        for(int64_t i = 0; i < num_tris * 3; ++i) {
+            int64_t new_idx = index_old_to_new[indices_ptr[i]];
+            indices_ptr[i] = Tindex_t(new_idx);
+        }
+        //Update triangle indices, no need to update other attributes.
+        tris = tris_cpu.To(mesh.GetDevice());
+        mesh.SetTriangleIndices(tris);
+    }
+
+    utility::LogDebug(
+             "[RemoveDuplicatedVertices] {:d} vertices have been removed.",
+             (int)(orig_num_vertices - k));
+
+    return mesh;
+
+}
+
+} //Anonymous namespace
+
+///
+/// Remove duplicate vertices from a mesh and return the resulting mesh.
+///
+TriangleMesh& TriangleMesh::RemoveDuplicateVertices()
+{
+    /// This function mainly does some checks and then calls the RemoveDuplicateVerticesWorker
+    /// with appropriate data type.
+    if(!HasVertexPositions()) {
+        utility::LogWarning("TriangeMesh::RemoveDuplicateVertices: No vertices present, ignoring.");
+        return *this;
+    }
+    auto vertices = GetVertexPositions();
+    auto triangles = GetTriangleIndices();
+    if (core::Int32 != triangles.GetDtype() && core::Int64 != triangles.GetDtype()) {
+        utility::LogError("Only Int32 or Int64 are supported for triangle indices");
+        return *this;
+    }
+    if (core::Float32 != vertices.GetDtype() && core::Float64 != vertices.GetDtype()) {
+        utility::LogError("Only Float32 or Float64 is supported for vertex coordinates");
+        return *this;        
+    }
+    
+    if(core::Float32 == vertices.GetDtype()) {
+        if (core::Int32 == triangles.GetDtype()) {
+            return RemoveDuplicateVerticesWorker<float, int>(*this);
+        } else {
+            return RemoveDuplicateVerticesWorker<float, int64_t>(*this);
+        }
+    } else {
+        if (core::Int32 == triangles.GetDtype()) {
+            return RemoveDuplicateVerticesWorker<double, int>(*this);
+        } else {
+            return RemoveDuplicateVerticesWorker<double, int64_t>(*this);
+        }
+    }
```

Comment:
While working on my `SelectByIndex` method, I discovered a set of macros to do this in more compact way:
https://github.com/isl-org/Open3D/blob/main/cpp/open3d/core/Dispatch.h#L90
I think we could rewrite it as smt like:
```suggestion
    DISPATCH_FLOAT_INT_DTYPE_TO_TEMPLATE(vertices.GetDtype(), triangles.GetDtype()), [&]() {
        RemoveDuplicateVerticesWorker<scalar_t, int_t>(*this);
    });
```